pyresample/geometry.py

```diff
@@ -1275,54 +1299,64 @@ def get_lonlats(self, nprocs=None, data_slice=None, cache=False, dtype=None):
             Calculate only coordinates for specified slice
         cache : bool, optional
             Store result the result. Requires data_slice to be None
+        dtype : numpy.dtype, optional
+            Data type of the returned arrays
+        chunks: int or tuple, optional
+            Create dask arrays and use this chunk size
 
         Returns
         -------
         (lons, lats) : tuple of numpy arrays
             Grids of area lons and and lats
         """
 
+        if cache:
+            warnings.warn("'cache' keyword argument will be removed in the "
+                          "future and data will not be cached.", PendingDeprecationWarning)
         if dtype is None:
             dtype = self.dtype
 
-        if self.lons is None or self.lats is None:
-            # Data is not cached
-            if nprocs is None:
-                nprocs = self.nprocs
-
-            # Proj.4 definition of target area projection
-            if nprocs > 1:
-                target_proj = Proj_MP(**self.proj_dict)
-            else:
-                target_proj = Proj(**self.proj_dict)
-
-            # Get coordinates of local area as ndarrays
-            target_x, target_y = self.get_proj_coords(
-                data_slice=data_slice, dtype=dtype)
-
-            # Get corresponding longitude and latitude values
-            lons, lats = target_proj(target_x, target_y, inverse=True,
-                                     nprocs=nprocs)
-            lons = np.asanyarray(lons, dtype=dtype)
-            lats = np.asanyarray(lats, dtype=dtype)
+        if self.lons is not None:
+            # Data is cache already
+            lons = self.lons
+            lats = self.lats
+            if data_slice is not None:
+                lons = lons[data_slice]
+                lats = lats[data_slice]
+            return lons, lats
+
+        # Get X/Y coordinates for the whole area
+        target_x, target_y = self.get_proj_coords(data_slice=data_slice, chunks=chunks, dtype=dtype)
+        if nprocs is None and not hasattr(target_x, 'chunks'):
+            nprocs = self.nprocs
+        if nprocs is not None and hasattr(target_x, 'chunks'):
+            # we let 'get_proj_coords' decide if dask arrays should be made
+            # but if the user provided nprocs then this doesn't make sense
+            raise ValueError("Can't specify 'nprocs' and 'chunks' at the same time")
```

Comment:
This might break quite a lot of legacy code, could we have a deprecation warning instead ?

Reply:
Do you mean legacy code within pyresample, satpy, or someone's operational code? As far as I know this shouldn't be an issue in satpy because we don't specify `nprocs` anywhere.